openmetadata-ui/src/main/resources/ui/src/components/Users/UsersProfile/UserProfileDetails/UserProfileDetails.component.tsx

```diff
@@ -171,27 +171,38 @@ const UserProfileDetails = ({
       ),
     [showChangePasswordComponent]
   );
-
   const handleChangePassword = async (data: ChangePasswordRequest) => {
     try {
       setIsLoading(true);
+      const newData = {
+        username: '',
+        requestType: RequestType.User,
+      };
+
+      if (isLoggedInUser) {
+        newData.username = userData.name;
+        newData.requestType = RequestType.Self;
+      } else {
+        newData.username = userData.name;
+      }
       const sendData = {
         ...data,
-        ...(isAdminUser &&
-          !isLoggedInUser && {
-            username: userData.name,
-            requestType: RequestType.User,
-          }),
+        ...newData,
       };
-      await changePassword(sendData);
-      setIsChangePassword(false);
-      showSuccessToast(
-        t('server.update-entity-success', { entity: t('label.password') })
-      );
+      await changePassword(sendData)
+        .then(() => {
+          showSuccessToast(
+            t('server.update-entity-success', { entity: t('label.password') })
+          );
+          setIsChangePassword(false);
+        })
+        .catch((error) => {
+          showErrorToast(error.response.data.message);
+        });
```

Comment:
await and then can't be used to gather, We have async await syntax for async operation in the code